From 03d63a1c3f70d651ffd31925dc2f0392c5513965 Mon Sep 17 00:00:00 2001
From: Takashi Iwai <tiwai@suse.de>
Date: Tue, 28 Jun 2011 13:11:33 +0200
Subject: [PATCH] Fix build with the new libnotify

The new libnotify has no longer notify_notification_new_with_status_icon().

Signed-off-by: Takashi Iwai <tiwai@suse.de>
---
 src/client.c |   11 +++++++++++
 1 files changed, 11 insertions(+), 0 deletions(-)

diff --git a/src/client.c b/src/client.c
index f992338..81f564d 100644
--- a/src/client.c
+++ b/src/client.c
@@ -36,6 +36,10 @@
 #include <gtk/gtk.h>
 #include <libnotify/notify.h>
 
+#ifndef NOTIFY_CHECK_VERSION
+#define NOTIFY_CHECK_VERSION(x, y, z)   0
+#endif
+
 #include <dbus/dbus.h>
 #include <dbus/dbus-glib.h>
 #include <dbus/dbus-glib-lowlevel.h>
@@ -194,11 +198,18 @@ init_state_change_notification (DriveGuard *drive_guard)
 
   drive_guard->current_notification = -1;
   for (n = 0; n < 2; n++) {
+#if NOTIFY_CHECK_VERSION(0, 7, 0)
+    drive_guard->notifications [n] =
+    notify_notification_new (gettext (message [n].summary),
+                             gettext (message [n].body),
+                             message [n].icon);
+#else
     drive_guard->notifications [n] =
     notify_notification_new_with_status_icon (gettext (message [n].summary),
                                               gettext (message [n].body),
                                               message [n].icon,
                                               drive_guard->status_icon);
+#endif
     notify_notification_set_urgency (drive_guard->notifications [n], message [n].urgency);
     notify_notification_set_timeout (drive_guard->notifications [n], message [n].timeout);
     /* first show to let system caching... */
-- 
1.7.6

